trp_use_case/model/use_case.py

```diff
@@ -59,18 +59,19 @@ def _get_use_case_hours(self):
             values = {
                 'hours': 0.0,
                 'hours_optional': 0.0,
+                'hours_nonoptional': 0.0,
             }
             for workload in use_case.workload_ids:
                 values['hours'] += workload.hours
                 if workload.optional:
                     values['hours_optional'] += workload.hours
                     values['hours_nonoptional'] = (
                         values['hours'] - values['hours_optional'])
-                    # TODO: why doesn't this work?
-                    # use_case.write(values)
-                    use_case.hours = values['hours']
-                    use_case.hours_optional = values['hours_optional']
-                    use_case.hours_nonoptional = values['hours_nonoptional']
+            # TODO: why doesn't this work?
+            # use_case.write(values)
```

Comment:
by now this is clear: We're in a compute function, there we don't want to actually write to the database, but only update the cache. The correct function to use here would be `.update()` (https://github.com/OCA/OCB/blob/8.0/openerp/models.py#L5465), but as the assignments don't trigger update statements, it doesn't really matter